### PR TITLE
Improve GitHub release discovery & other stuff

### PR DESCRIPTION
**Pull request by jtojnar**

### CODE REVIEW

File: src/GH.hs [hunk 1]
```diff
@@ -44,20 +52,33 @@ data URLParts = URLParts
...
   }
 
+-- | Parse owner-repo-branch triplet out of URL
+-- We accept URLs pointing to uploaded release assets
+-- that are usually obtained with fetchurl, as well
+-- as the generated archives that fetchFromGitHub downloads.
+--
+-- Examples:
+--
+-- >>> runExceptT (parseURL "https://github.com/blueman-project/blueman/releases/download/2.0.7/blueman-2.0.7.tar.xz")
```
Comment (jtojnar) [inline src/GH.hs]: I am not really sure how to test this. We probably have to split the function and operate on `Maybe`.

File: src/Utils.hs [hunk 2]
```diff
@@ -74,12 +74,15 @@ runtimeDir = do
...
-setupNixpkgs :: IO ()
-setupNixpkgs = do
+setupNixpkgs :: Options -> IO ()
+setupNixpkgs o = do
   fp <- getUserCacheDir "nixpkgs"
   exists <- doesDirectoryExist fp
   unless exists $ do
-    _ <- shelly $ run "hub" ["clone", "nixpkgs", T.pack fp] -- requires that user has forked nixpkgs
+    _ <-
+      shelly $ do
```
Comment (jtojnar) [inline src/Utils.hs]: Maybe we need to untangle the modules to be able to use `ourShell`.

File: test/doctests.hs [hunk 3]
```diff
@@ -1,4 +1,4 @@
 import Test.DocTest
 
 main :: IO ()
-main = doctest ["-isrc", "-XOverloadedStrings", "src/Utils.hs"]
+main = doctest ["-isrc", "-XOverloadedStrings", "src/Version.hs", "src/GH.hs"]
```
Comment (jtojnar) [inline test/doctests.hs]: It would be nice to make the file list dynamic.

Comment (jtojnar) [reply test/doctests.hs]: Hmm, apparently, it should be possible to just list `src/` https://hackage.haskell.org/package/doctest-0.16.0.1/docs/Test-DocTest.html#v:doctest

Comment (jtojnar) [reply test/doctests.hs]: But it seems very slow, even with the `--fast` argument.

File: src/Utils.hs (hunk 2 above)
Comment (ryantm) [reply src/Utils.hs]: I'm generally trying to push all the shelly invocations to the lowest level IO functions, and us regular IO for setting up the environment and working directory stuff. Basically, I'm trying to move to only using shelly to directly call a shell command and nothing else fancy.

**Pull request merged**